core/src/main/scala/org/apache/spark/sql/rapids/tool/qualification/QualificationAppInfo.scala

```diff
@@ -454,6 +454,19 @@ class QualificationAppInfo(
         _.unsupportedExprs)).flatten.filter(_.nonEmpty).toSet.mkString(";")
         .trim.replaceAll("\n", "").replace(",", ":")
 
+      // Get all unsupported execs and expressions from the plan in form of map[exec -> exprs]
+      val unsupportedExecExprsMap = planInfos.map { p =>
+        val topLevelExecs = p.execInfo.filterNot(_.isSupported).filterNot(
+          x => x.exec.startsWith("WholeStage"))
+        val childrenExecs = p.execInfo.flatMap { e =>
+          e.children.map(x => x.filterNot(_.isSupported))
+        }.flatten
+        val execs = topLevelExecs ++ childrenExecs
+        val exprs = execs.filter(_.unsupportedExprs.nonEmpty).map(
+          e => e.exec -> e.unsupportedExprs.mkString(";")).toMap
+        exprs
+      }.flatten.toMap
```

Comment:
nit: replace `.map` and `.flatten` with `.flatMap`

Reply:
I wonder how many rows will be generated by a large eventlog.
I have a couple of questions:
- I see that we use list/arrays instead of sets. is it possible to have multiple rows showing the same values? or that's already taken care of by how the plan is parsed?
- do we need to limit the number of rows per application?

Reply:
Removed .flatten and updated top level map to flatMap.

For a given application, all rows in the output file are distinct. Verified it by creating multiple reads of the same file for an application. I am not certain if we need to limit the number of rows per application. Since it includes only those Execs/exprs which are not supported, the number of output rows per application isn't large from testing.